src/server.rs

```diff
@@ -1708,24 +1707,23 @@ impl Drop for ActiveInfo {
         let mut info = self.info.borrow_mut();
         info.active -= 1;
         if info.active == 0 {
-            if let Some(task) = info.blocker.take() {
-                task.notify();
+            if let Some(waker) = info.waker.take() {
+                waker.wake();
             }
         }
     }
 }
 
-impl Future for WaitUntilZero {
-    type Item = ();
-    type Error = io::Error;
+impl std::future::Future for WaitUntilZero {
+    type Output = io::Result<()>;
 
-    fn poll(&mut self) -> Poll<(), io::Error> {
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> std::task::Poll<Self::Output> {
```

Comment:
I guess at some future point we should make this return `std::task::Poll<()>`, because we never care about the error case of `io::Result`?  Or am I missing something about futures-based code?

Reply:
It's because of the compat stuff. To get an impl Future on Compat, you need an impl TryFuture, which you get if you have an impl Future with a Result Output.

Reply:
Fair enough!